universal-application-tool-0.0.1/app/services/cloud/azure/BlobStorageUploadRequest.java

```diff
@@ -0,0 +1,89 @@
+package services.cloud.azure;
+
+import com.google.auto.value.AutoValue;
+import services.cloud.StorageServiceName;
+import services.cloud.StorageUploadRequest;
+
+/** This holds the information needed to upload a blob storage request. */
+@AutoValue
+public abstract class BlobStorageUploadRequest implements StorageUploadRequest {
+
+  public abstract String accountName();
+
+  public abstract String containerName();
+
+  public abstract String fileName();
+
+  public abstract String sasToken();
+
+  public abstract String blobUrl();
+
+  public abstract String successActionRedirect();
+
+  @Override
+  public abstract String serviceName();
+
+  public static Builder builder() {
+    return new AutoValue_BlobStorageUploadRequest.Builder()
+        .setServiceName(StorageServiceName.AZURE_BLOB.getString());
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * Get file name for the blob. This is used to build the canonicalized resource part of the
+     * signature string.
+     */
+    abstract String fileName();
```

Comment:
question (blocking): why are getters on the builder?

Reply:
because we used the other AWS builder as a template originally, and then changed the implementation without removing the getters. they're removed now.

Reply:
Looks like they're still here?

Reply:
For some reason, sometimes github doesn't show the latest version by default. you'll have to go to this menu on the top left and choose "show all changes"
![image](https://user-images.githubusercontent.com/19631367/146984473-78047042-fc89-4b3d-a46b-75c0c26bad04.png)